tests/structures/test_class.py

```diff
@@ -46,6 +46,23 @@ def __str__(self):
             print('Done.')
             """, run_in_function=False)
 
+    # @unittest.expectedFailure
+    def test_getattr(self):
+        self.assertCodeExecution("""
+            class MyClass:
+                def __init__(self):
+                    self.x = 1
+                    self.y = 22
+                def __getattr__(self, attr):
+                    self.y = 42
+                    return y+1
+
+            obj = MyClass()
+            print(obj.x)
+            print(obj.fail)
+            print(obj.y)
+        """)
+
```

Comment:
This test will pass; but it's not a good test. If you actually run this, you'll get exceptions because the `y` on line 58 is an unknown variable. The error will be consistent between Python and Batavia, so the test passes; but it's not fully testing like you think it is.

It also doesn't test `__getattribute__`, which was the major contribution of this patch.